packages/block-editor/src/components/writing-flow/index.js

```diff
@@ -505,11 +506,15 @@ export default function WritingFlow( { children } ) {
 		}
 	}, [ hasMultiSelection, isMultiSelecting ] );
 
+	const className = classnames( 'block-editor-writing-flow', {
+		'is-navigate-mode': isNavigationMode,
+	} );
```

Comment:
I'm wondering if this is the right place for it.

If this is applied at this level, should the navigation styles be moved to the WritingFlow as well? For me ideally, this shouldn't be applied here unless the navigation mode is triggered here as well.

If the navigation mode is triggered from the block list (talking about the shortcuts), then maybe the style should stay where it is now but the class should also move to the BlockList.

Reply:
Which shortcut are you talking about? This component contains almost all logic for navigation mode (except for the breadcrumb to go back into Edit mode). It's used as the container to control the mode, so I feel like this component is appropriate for the class as well. I do agree that perhaps the CSS selectors should be moved as well, but I wasn't 100% sure about it.

Reply:
I'm talking about "Enter" to enter "edit" mode and "Escape" to enter "navigation" mode.

Reply:
@ellatrix As an aside, I have a concern about all these refactorings we're doing to make the block.js smaller (classes and behavior). We often think about the Gutenberg use case in Core but we forget/neglect the packages and their reusable aspect. We should be making more playground stories to ensure we don't regress in these use cases.

Reply:
`Escape` to enter navigation mode is added by `WritingFlow`. `Enter` to enter edit mode is added by the breadcrumb within the block toolbar. Since `WritingFlow` forms a container around the blocks, it seems the most appropriate for the class (as opposed to the breadcrumb). I feel like the block and block list components don't have much to do with the mode other than some minor styling difference (which will even be removed in the future it seems).

Reply:
I don't think I'm neglecting the packages here. Previously `WritingFlow` already handled a lot of navigation mode, so without it, it would be broken.

Reply:
I'm not blaming you or anyone :)

I know I have a tendency to forget about the playground and third-party usage sometimes and when I see things that could break that in other PRs, I wonder if I might not be the only one.

Reply:
I know :) It's good to raise this. Perhaps we should add some tests for using the block editor on its own. I'll have a look into it at some point. That said, I don't think this PR in particular would break anything. If `BlockList` is used on its own, you can't really use navigation mode with `WritingFlow`. If `WritingFlow` is added, the class will be there as well.